api-call.py

```diff
@@ -1,29 +1,334 @@
-        print("sucessfully fetched the data with parameters")
-        format_result(response.json())
-    else:
-        print(
-            f"There's a {response.status_code} error with this request")
+        # Configure the logger
+        logging.basicConfig(
+            filename='logfile.log',
+            level=logging.INFO,
+            format='%(asctime)s [%(levelname)s] %(message)s',
+            datefmt='%Y-%m-%d %H:%M:%S %Z'
+        )
 
-def format_result(data):
-    text = json.dumps(data, sort_keys=True, indent=4)
-    print(text)
+        logging.info(f'{current_time}: {message}')
 
 
+    def get_access_token(self) -> str:
+        """Get access token from OAuth2 for connecting with OSU public APIs
 
+        :returns: The access token string
+        """
+        url = self.config['access_token']['url']
+        payload = self.config['access_token']['payload']
+        payload['grant_type'] = 'client_credentials'
+        headers = {'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8'}
 
-if __name__ == "__main__":
-    parameters = {
-            
-        }
+        response = requests.post(
+            url,
+            data=payload,
+            headers=headers,
+            timeout=10
+        )
 
-    header={"Content-Type": "application/x-www-form-urlencoded;charset=utf-8" , "Authorization":"Bearer CUTOuxWmuRfKQKynBGjFGmBROM9R"}
+        if response.status_code == 200:
+            access_token = response.json()['access_token']
+            access_token_extended = f'Bearer {access_token}'
+            return access_token_extended
+        else:
+            self.log_message(f'Request failed with status code: {response.status_code}')
+            return ''
+
+
+    def get_api_data(self, url: str, parameters: Dict[str, Any], header: Dict[str, str]) -> Any:
+        """Get Request for getting the data from API
+
+        :param url: URL of the API
+        :param parameters: The parameters for API
+        :param header: The headers for API with authorization
+        :return: The response data from the API
+        """
+
+        response = requests.get(
+            url,
+            params=parameters,
+            headers=header,
+            timeout=10
+        )
+        if response.status_code == 200:
+            response_data = self.format_result(response)
+            return response_data
+        else:
+            print(response.status_code)
+            generated_error = f'There is a {response.status_code} error with this request'
+            self.log_message(generated_error)
+
+
+    def format_result(self, response: requests.Response) -> Any:
+        """Returns data from response object after JSON validation
+
+        :param response: The API response object
+        :return: The validated data
+        """
+        try:
+            data = response.json()
+        except json.decoder.JSONDecodeError:
+            print("Invalid JSON response")
+
+        text = json.dumps(data, indent=4)
```

Comment:
What's `text` for?

Reply:
it's for checking the output of API as a whole, used for testing the response.